01-JS-DOM-APIs/js/index.js

```diff
@@ -0,0 +1,388 @@
+        if (Clicks == callId){ //Is it the last call?
+            console.log("HERE" + Clicks);
+            showJoke(req); // This function it's the one that show JOKES. It's used by exercises 6 and 7.
+            changeColor("green");
+            changeAdvice("JOKE SUCCESS!!");
+            JSuccess = true;
+        }
+    });
+    req.send(null); // send the HttpRequest.
+}
+
+/* EX 7 (KEY) and 6
+ * function showJoke(reqJSON)
+ * where reqJSON -> It's a JSON object
+ *  This function takes a JSON answer of the HttpRequest, and show it in the HTML Document, modifying the DOM.
+*/
+function showJoke (reqJSON){
+    if (reqJSON.status < 400){ // if < 400 ->> Success!
+        var reqParse = JSON.parse(reqJSON.responseText); // parse the Json answer
+             /*Different commands i tried to understand the structure of the requested JSON object -
+             * i don't delete it in case somebody seeing the code wanna to know its structure.
+                 console.log(reqJSON.status, reqJSON.statusText); // Req status
+                 console.log(reqJSON.responseText); // Devolution
+                 console.log(reqParse); // [type (string), value (object)]
+                 console.log(reqParse.value); // [id (integer), joke(string), categories(array)]
+                 console.log(reqParse.value.categories); // categories elements
+             */
+        var elem = document.createElement("p"); // new element <p>
+        var cont = document.createTextNode(reqParse.value["joke"]); // new element <p> content
+        elem.appendChild(cont); // insert node cont in node elem <p>
+        var where = document.getElementById("hide"); // looking for the section where i will work
+        removeAllChildNodes(where); // this function remove all childs of a node, in this case, it will delete all the content
+        where.appendChild(elem.appendChild(cont)); // insert the <p> element with its content in the section "hide"
+        where.style.fontSize= "20px"; // Some style for a better watching
+        where.style.fontWeight = "bold";
+    }
+}
+
+/* EX 7 (KEY) The CONFIG objects constructor*/
+function CONFIG() {
+    this.action = "GET"; //method of the AJAX call. Default = "GET"
+    this.url = ""; //url of the AJAX call. Default = ""
+    this.async = false; //asynchronous or synchronous. Default = false (synchronous)
+    this.user = ""; //user of the AJAX call. Default = ""
+    this.password = ""; //password of the AJAX call. Default = ""
+    this.hidePassword = true; //this option, say to the object, if when you call de info() method,
+                              // it will show the password or not.
+    this.send = null; // what to send. Default = "". For a "GET" AJAX call, this value it's not use.
+    function writePassword (hide , pass){ //This method change the password for *, in case hidePassword it's set to true.
+        var text = pass;
+        if (hide) {
+            text = "";
+            var i = 0;
```

Comment:
try to use more semantic names for variables, what does `i` mean? for this case is fine but when you have to be descriptive to understand what is the while or for doing you may need to review the variables names.